src/mouse_pos.rs

```diff
@@ -2,42 +2,65 @@ use std::{fmt::Display, ops::Deref};
 
 use bevy::prelude::*;
 
-use crate::MouseTrackingSystem;
-
 /// Plugin that tracks the mouse location.
-#[non_exhaustive]
 pub enum MousePosPlugin {
-    /// Track the mouse without transforming it to worldspace.
-    None,
-    /// Transform the mouse position into worldspace, using an orthographic camera.
-    Orthographic,
+    /// Configuration for apps that have a single main camera.
+    /// Provides global Resources for [`MousePos`] and [`MousePosWorld`].
+    SingleCamera,
+    /// Configuration for apps that have multiple cameras which must be handled separately.
+    MultiCamera,
 }
 
 impl Plugin for MousePosPlugin {
     fn build(&self, app: &mut App) {
-        app.insert_resource(MousePos::default());
+        #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, SystemLabel)]
+        enum MouseSystem {
+            ScreenPos,
+            WorldPos,
+        }
+
+        // System to add mouse tracking components.
+        // Runs once at the end of each frame. This means that no cameras will have
+        // mouse tracking components until after the first frame.
+        // This might cause some issues, but it's probably for the best since,
+        // during the first frame, nothing has been rendered yet.
+        app.add_system_to_stage(CoreStage::PostUpdate, add_pos_components);
+
+        #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, StageLabel)]
+        struct MouseStage;
+
+        app.add_stage_before(
+            CoreStage::PreUpdate,
+            MouseStage,
+            SystemStage::single_threaded(),
+        );
+
+        app.add_system_to_stage(MouseStage, update_pos.label(MouseSystem::ScreenPos));
         app.add_system_to_stage(
-            CoreStage::First,
```

Comment:
I'm curious what is driving this change?

Reply:
At some point I was having a lot of trouble with system execution order and it was simpler to just add a new stage for this plugin. I know that stages limit parallelization in bevy so I'll see if I can move them back into `CoreStage::First` without breaking everything.